scripts/globals/combat/physical_utilities.lua

```diff
@@ -684,3 +684,291 @@ end
+                return math.max(0, blockRate + skillModifier)
+            -- mobs and trusts use max skill for job and level
+            elseif defender:isTrust() then
+                -- TODO: check trust type for ilvl > 99 when implemented
+                blockSkill = defender:getMaxSkillLevel(math.min(defender:getMainLvl(), 99), defender:getMainJob(), xi.skill.SHIELD)
+            else
+                blockSkill = defender:getMaxSkillLevel(defender:getMainLvl(), defender:getMainJob(), xi.skill.SHIELD)
+            end
+        else -- No block mobmod so zero rate
+            return 0
+        end
+    end
+
+    if defender:isPC() then
+        if shieldSize == 1 then -- Buckler
+            blockRate = 55
+        elseif shieldSize == 2 then -- Round
+            blockRate = 40
+        elseif shieldSize == 3 then -- Kite
+            blockRate = 45
+        elseif shieldSize == 4 then -- Tower
+            blockRate = 30
+        elseif shieldSize == 5 then -- Aegis and Srivatsa
+            blockRate = 50
+        elseif shieldSize == 6 then -- Ochain https://www.bg-wiki.com/ffxi/Category:Shields
+            blockRate = 108
+        else
+            return 0
+        end
+    end
+
+    -- Check for Reprisal and adjust skill and block rate bonus multiplier
+    if defender:hasStatusEffect(xi.effect.REPRISAL) then
+        blockSkill   = blockSkill * 1.15
+        reprisalMult = 1.5
+
+        -- Adamas and Priwen set the multiplier to 3.0x while equipped
+        if defender:getMod(xi.mod.REPRISAL_BLOCK_BONUS) > 0 then
+            reprisalMult = 3.0
+        end
+    end
+
+    skillModifier = (blockSkill - attackSkill) * 0.2325
+
+    -- Add skill and Palisade bonuses and multiply by Reprisals bonus
+    blockRate = (blockRate + skillModifier + palisadeMod) * reprisalMult
+
+    -- Apply the lower and upper caps
+    blockRate = utils.clamp(blockRate, 5, 100)
+
+    return blockRate
+end
+
+xi.combat.physical.handleBlock = function(defender, attacker, damage)
+    if
+        xi.combat.physical.canBlock(defender, attacker) and
+        xi.combat.physical.calculateBlockRate(defender, attacker) > math.random(100)
+    then
+        if defender:isPC() then
+            local absorb = 100
+            absorb = utils.clamp(absorb - defender:getShieldAbsorptionRate(), 0, 100)
+            absorb = absorb + defender:getMod(xi.mod.SHIELD_DEF_BONUS)
+            damage = math.floor(damage * (absorb / 100))
+            defender:trySkillUp(xi.skill.SHIELD, attacker:getMainLvl())
+        else
+            damage = math.floor(damage * 0.5)
+        end
+    end
+
+    return damage
+end
+
+xi.combat.physical.isParried = function(defender, attacker)
+    local parried = false
+    if
+        xi.combat.physical.canParry(defender, attacker) and
+        xi.combat.physical.calculateParryRate(defender, attacker) > math.random(100)
+    then
+        parried = true
+        if defender:isPC() then
```

Comment:
can you add a TODO to implement this Turms mod
![image](https://github.com/LandSandBoat/server/assets/60417494/ebe48587-6c75-481d-84a5-43880c5b89ee)

Reply:
> can you add a TODO to implement this Turms mod ![image](https://private-user-images.githubusercontent.com/60417494/335828488-ebe48587-6c75-481d-84a5-43880c5b89ee.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTczMjI4ODgsIm5iZiI6MTcxNzMyMjU4OCwicGF0aCI6Ii82MDQxNzQ5NC8zMzU4Mjg0ODgtZWJlNDg1ODctNmM3NS00ODFkLTg0YTUtNDM4ODBjNWI4OWVlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjAyVDEwMDMwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFmNzg4ZGI3NjFlNWRhNjI4ZDgxY2U4NGY5ZjE4YjUwMzhiYzE0OGE5YTMxNGE3YWRiMWI4MGIxYjgyNmIxMTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.pK4Q6LEroCjdd1tu6nj9SVxdUaGYPZJ6fQVGAmZkUGU)

Yep, I added a TODO, thanks!